[MPIR:x64-linux] build failure (static linking)

**Host Environment**
- OS: Linux (Ubuntu 20)
- Compiler: g++ 9.3

**To Reproduce**
Steps to reproduce the behavior:
```sh
./vcpkg install cgal eigen3
git clone https://github.com/amakhno/vcpkg-gmp-bug-example.git
cd vcpkg-gmp-bug-example
cmake . && make
```

**Failure logs** 
vcpkg install works perfectly but I have an error during the project build (linking)
```log
-- Using header-only CGAL
-- Targetting Unix Makefiles
-- Using /usr/bin/c++ compiler.
-- Boost include dirs: /home/noir/projects/vcpkg/installed/x64-linux/include
-- Boost libraries:    
-- Using gcc version 4 or later. Adding -frounding-math
CMake Warning at /home/noir/projects/vcpkg/installed/x64-linux/share/cgal/CGAL_enable_end_of_configuration_hook.cmake:99 (message):
  =======================================================================

  CGAL performance notice:

  The variable CMAKE_BUILD_TYPE is set to "".  For performance reasons, you
  should set CMAKE_BUILD_TYPE to "Release".

  Set CGAL_DO_NOT_WARN_ABOUT_CMAKE_BUILD_TYPE to TRUE if you want to disable
  this warning.

  =======================================================================
Call Stack (most recent call first):
  CMakeLists.txt:9999 (CGAL_run_at_the_end_of_configuration)


-- Configuring done
-- Generating done
-- Build files have been written to: /home/noir/projects/vcpkg-bug-example
[ 50%] Linking CXX shared library libexample.so
/usr/bin/ld: /home/noir/projects/vcpkg/installed/x64-linux/debug/lib/libgmp.a(modexact_1c_odd.o): relocation R_X86_64_PC32 against symbol `__gmp_modlimb_invert_table' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/example.dir/build.make:86: libexample.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/example.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

**Additional context**
There is no such a problem when you try to buld it with a dynamic library or a console application. 
I've tried to reinstall vcpkg completely but it didn't help. 
The same project works like a charm with :windows-x64-static on my Windows machine.

As I googled, it's likely that linux default triplet is x64-linux which is static, but I couln't find any reasons why it doesn't work

The problem is probably in MPIR but I'm not sure

@PhoebeHui , 
Thank you for your reply!

Do have any ideas on how to deal with this problem? 
I tried to look inside the .a file and there are dozens of links but it still doesn't work.

If I change the MPIR portfile.cmake to: (add -fPIC)

```cmake
string(APPEND VCPKG_CXX_FLAGS " -Wno-implicit-function-declaration -fPIC")
```

It changes nothing

So I also changed the dependencies of MPFR and CGAL to GMP but there is no real difference in the result

```
/bin/ld: /home/noir/projects/vcpkg/installed/x64-linux/debug/lib/libgmp.a(bdiv_q_1.o): relocation R_X86_64_PC32 against symbol `__gmp_binvert_limb_table' can not be used when making a shared object; recompile with -fPIC
/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/example.dir/build.make:86: libexample.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/example.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

--- a/ports/gmp/portfile.cmake
+++ b/ports/gmp/portfile.cmake
@@ -49,6 +49,7 @@ vcpkg_configure_make(
     OPTIONS
         ${OPTIONS}
         --enable-cxx
+        --with-pic
 )
 
 set(tool_names bases fac fib jacobitab psqr trialdivtab)

--- a/ports/gmp/vcpkg.json
+++ b/ports/gmp/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gmp",
   "version": "6.2.1",
-  "port-version": 9,
+  "port-version": 10,
   "description": "The GNU Multiple Precision Arithmetic Library",
   "homepage": "https://gmplib.org",
   "supports": "!(windows & (arm | arm64))",

--- a/ports/mpir/portfile.cmake
+++ b/ports/mpir/portfile.cmake
@@ -19,12 +19,12 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     vcpkg_find_acquire_program(YASM)
 
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-        set(SHARED_STATIC "--enable-static --disable-shared")
+        set(SHARED_STATIC --enable-static --disable-shared --with-pic)
     else()
-        set(SHARED_STATIC "--disable-static --enable-shared")
+        set(SHARED_STATIC --disable-static --enable-shared)
     endif()
 
-    set(OPTIONS "--disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC}")
+    set(OPTIONS --disable-silent-rules --enable-gmpcompat --enable-cxx ${SHARED_STATIC})
 
     string(APPEND VCPKG_C_FLAGS " -Wno-implicit-function-declaration")
     string(APPEND VCPKG_CXX_FLAGS " -Wno-implicit-function-declaration")

--- a/ports/mpir/vcpkg.json
+++ b/ports/mpir/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "mpir",
   "version-date": "2022-03-02",
+  "port-version": 1,
   "description": "Multiple Precision Integers and Rationals",
   "homepage": "https://github.com/wbhart/mpir",
   "license": "GPL-3.0-only",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2606,7 +2606,7 @@
     },
     "gmp": {
       "baseline": "6.2.1",
-      "port-version": 9
+      "port-version": 10
     },
     "gmsh": {
       "baseline": "4.9.0",
@@ -4686,7 +4686,7 @@
     },
     "mpir": {
       "baseline": "2022-03-02",
-      "port-version": 0
+      "port-version": 1
     },
     "mpmcqueue": {
       "baseline": "2021-12-01",

--- a/versions/g-/gmp.json
+++ b/versions/g-/gmp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f4748213535c3fd004de44f6b1f15d123927cce6",
+      "version": "6.2.1",
+      "port-version": 10
+    },
+    {
       "git-tree": "7b9a71843073bf4a86bb64ddf219c9900ebb3dbd",
       "version": "6.2.1",
       "port-version": 9

--- a/versions/m-/mpir.json
+++ b/versions/m-/mpir.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9191f07cfaade82121abb4d37cb652182c0e55f6",
+      "version-date": "2022-03-02",
+      "port-version": 1
+    },
+    {
       "git-tree": "5358d4a724061eab499969ae3b56f8abbdea3347",
       "version-date": "2022-03-02",
       "port-version": 0

